enterprise/api/v1/serializers.py

```diff
@@ -214,3 +215,106 @@ def update(self, instance, validated_data):
     user = UserSerializer(read_only=True)
     enterprise_customer = EnterpriseCustomerSerializer(read_only=True)
     data_sharing_consent = UserDataSharingConsentAuditSerializer(many=True, read_only=True)
+
+
+class EnterpriseCourseCatalogReadOnlySerializer(serializers.Serializer):
+    """
+    Serializer for enterprise customer catalog.
+    """
+    # pylint: disable=invalid-name
+    id = serializers.IntegerField(read_only=True, help_text=_('Enterprise course catalog primary key.'))
+    name = serializers.CharField(help_text=_('Catalog name'))
+    query = serializers.CharField(help_text=_('Query to retrieve catalog contents'))
+    courses_count = serializers.IntegerField(read_only=True, help_text=_('Number of courses contained in this catalog'))
+    viewers = serializers.ListField(
+        allow_null=True, allow_empty=True, required=False,
+        help_text=_('Usernames of users with explicit access to view this catalog'),
+        style={'base_template': 'input.html'},
+        child=serializers.CharField(),
+    )
+
+    def create(self, validated_data):
+        """
+        Do not perform any operations for state changing requests.
+        """
+        pass
+
+    def update(self, instance, validated_data):
+        """
+        Do not perform any operations for state changing requests.
+        """
+        pass
+
+
+class EnterpriseCatalogCoursesReadOnlySerializer(serializers.Serializer):
+    """
+    Serializer for enterprise customer catalog courses.
+    """
+    count = serializers.IntegerField(read_only=True, help_text=_('Total count of catalog courses.'))
+    next = serializers.CharField(read_only=True, help_text=_("URL to fetch next page of courses."))
+    previous = serializers.CharField(read_only=True, help_text=_("URL to fetch previous page of courses."))
+    results = serializers.ListField(read_only=True, help_text=_("list of courses."))
+
+    def update_enterprise_courses(self, request):
+        """
+        This method adds enterprise specific metadata for each course.
+
+        We are adding following field in all the courses.
+            tpa_hint: a string for identifying Identity Provider.
+        """
+        courses = []
+        tpa_hint = utils.get_identity_provider_id_for_user(request.user)
```

Comment:
I'm not sure how I feel about deriving this based on "session" information that's basically hidden - it conflates the concepts of "the user who can retrieve a piece of information" and "the user who can use a piece of information". I'd rather split out "enterprise_customer_uuid" or something similar into the URL, and then derive the tpa_hint based on the URL, with appropriate access restrictions. If we switch to fetching based on EC UUID, as suggested below, rather than on catalog ID, then we get that for free - we can pull the relevant course catalog data for a given customer, and also apply customer-specific transforms to it.